SpatialGDK/Source/SpatialGDK/SpatialGDK.Build.cs

```diff
@@ -15,7 +15,20 @@ public SpatialGDK(ReadOnlyTargetRules Target) : base(Target)
     {
         bLegacyPublicIncludePaths = false;
         PCHUsage = ModuleRules.PCHUsageMode.UseExplicitOrSharedPCHs;
-        bUseUnity = false;
+        
+        // Buildkite can store environment variables like "True" and "False" as "1" and "0", which can't be parsed by bool.TryParse() or bool.Parse()
+		string unity_mode_env = Environment.GetEnvironmentVariable("GDK_UNITY_MODE");
+		bool unity_mode;
+        if (bool.TryParse(unity_mode_env, out unity_mode)) {
+            // unity_mode_env is "True", "False" or case-insensitive versions of these
+    		bUseUnity = unity_mode;
+        } else if (unity_mode_env.Equals("0")) {
+            bUseUnity = false;
+        } else if (unity_mode_env.Equals("1")) {
+            bUseUnity = true;
+        } else {
+            throw new FormatException("Value of GDK_UNITY_MODE environment variable was not recognized as a valid Boolean.");
+        }
```

Comment:
I think this could be refactored into a function and a small Util.cs file?
Also, what happens if you don't have the envvar set, would we get the format exception?
(We wouldn't be able to build locally then)

Reply:
agreed, will refactor. also yes, if building locally (not through `UnrealGDKBuild`'s `run_local.py()`) and the env vars are not set, then the exception will be thrown; perhaps it would be better to default to non-unity builds and not bother with the exception?

Reply:
Yes ^